### DISCUSSION TITLE
How to handle DynamoDB error?

### DISCUSSION THREAD

**Discussion by d2a4u** [tywalch/electrodb#316] [General]
Hi, ElectroDB returns `ElectroError` error type, however, all DynamoDB errors are under 4001 code. How can I distinguish different error types such as `ConditionalCheckFailedException`? For example, I use `put` to insert new item into the table with a conditional check to avoid duplicates.

**Comment by tywalch** [comment]
Hi @d2a4u 👋 

I just published an update (`2.10.4`) that makes this easier. The original error thrown by the DocumentClient you provided will now appear on the `cause` property of any ElectroError. You can also pass the error type expected for your given client as the first generic argument to ElectroError to strongly type the returned error beyond the base Error class.

**Reply by d2a4u** [reply]
That is amazing fast response and release. Thank you :)
